### ISSUE TITLE
Cant save via Web GUI

### ISSUE THREAD

**Issue by mjd515** [mylar3/mylar3#1179]
i cant get the web GUI to save it always displays an an error occurred. i can edit the file manually with a text editor, and the changes will replicate after a reboot. 

i am using docker/docker compose on linux to load the application, i am using the linuxserver nightly image 

i have tried chmoding the file to 777 as well as verifying my GUID and PUID is set correctly 

i have attached a link to the carepackage 

https://mega.nz/file/5pA0CKTC#4VcnU3N6LyTYBjqWvi0S0wYdy_UzPd7CmUQh0RutTpM

**Comment by DieselTech** [comment]
I'm not seeing anything stand out in the carepackage. You mention that it shows an error occurred when trying to save, what specifically is that error and do you happen to have screenshots of it?

**Comment by mjd515** [comment]
I can screen shot it but it literally just says an error occured in a red box at the bottom. And then there isn't anything in the logs even after turning on debug

**Comment by barbequesauce** [comment]
How about in the JavaScript console /developer tools?

**Comment by mjd515** [comment]
[mylar.mdaps.net-1650728723416.log](https://github.com/mylar3/mylar3/files/8547532/mylar.mdaps.net-1650728723416.log)
thats the error i get in the console i also have a text file of the out put of the erron from the URL as a stack trace as well as node js
[new 1.txt](https://github.com/mylar3/mylar3/files/8547537/new.1.txt)

**Comment by evilhero** [comment]
Check your nginx logs for an error. 

If I recall from another post along time ago (that I cannot find for the life of me), because Mylar sends the complete config in one url fetch, it can cause nginx to barf if the length is too long due to ``large_client_header_buffers`` being set to a lower than default option. 

You may have to specify that value for your mylar endpoint in your nginx config.

**Comment by mjd515** [comment]
i use traefik and not ngix unless the mylar app uses ngix in the background?

**Comment by barbequesauce** [comment]
No it doesn't. I believe there are similar config options in traefik...

**Comment by mjd515** [comment]
ok ill check it out. i didnt see any errors in traefik logs but im sure thats probably the case. thanks! ill close the issue if i figure it out or come back with findings if no change.

**Comment by evilhero** [comment]
Yeah my point being it's not really a Mylar issue as far as being able to save. There's some config option that needs to be enabled or disabled so that the large url can be saved proeprly from whatever proxy manager you're using. 

If you were to try and save the config outside of the reverse proxy, it would be able to save the config.

**Comment by barbequesauce** [comment]
This might help? https://community.traefik.io/t/431-request-header-fields-too-large/11047

**Comment by mjd515** [comment]
so the issue is authelia. idk why but when you save it send the whole header to authelia and authelia prevents it due to being too big. i just switch to basic auth for the mylar container and that fixed it

**Comment by Typhonragewind** [comment]
Alright, I figured it out with various clues left in these issues and i'm going to write this here in case someone else needs it. The issue in my case was indeed Authelia, which by default reads buffer sizes of only 4096 bytes.

So, if you don't want to deactivate Authelia for this (like me) you need to change the "read_buffer_size" entry in the Authelia settings to 8192 or larger ( i set mine at 16384) and restart it.

**Comment by mjd515** [comment]
> Alright, I figured it out with various clues left in these issues and i'm going to write this here in case someone else needs it. The issue in my case was indeed Authelia, which by default reads buffer sizes of only 4096 bytes.
> 
> So, if you don't want to deactivate Authelia for this (like me) you need to change the "read_buffer_size" entry in the Authelia settings to 8192 or larger ( i set mine at 16384) and restart it.

well this helps a lot! thanks so much for figuring out what to change!

**Comment by ColinHebert** [comment]
For what it's worth @evilhero, @barbequesauce, the root issue is that currently mylar makes a GET call to save the settings.
Given that all the settings are being saved, it would be significantly better to make a POST call.

The reason this goes through a GET (even though the form is properly set to run a POST), is that the config save is done via `doAjaxCall()` (https://github.com/mylar3/mylar3/blob/master/data/interfaces/default/config.html#L1715), which itself calls `$.ajax()` (https://github.com/mylar3/mylar3/blob/master/data/js/script.js#L195) but doesn't set a `method` (or `type` for JQuery < 1.9.0), therefore it defaults to a GET (https://api.jquery.com/Jquery.ajax/)

This behaviour should be modifiable and made easier for people to use mylar without having to change their reverse proxy or their SSO mechanisms.

**Comment by barbequesauce** [comment]
Great catch @ColinHebert thank you! I've never been able to replicate this scenario on my test rig, never figured out why. Anyway - I'm going to fork dev now with an added method: POST, as you describe; hopefully @mjd515 or someone else who can repro would be willing to switch branches and test it...
